src/backend/upload_file/upload_file_for_chunks.py

```diff
@@ -0,0 +1,207 @@
+                    if len(reader.pages) > 0:
+                        approaches.append("Approach 3: PDF content extracted")
+            except Exception:
+                approaches.append("Approach 3: PDF content extraction failed")
+
+        elif filename.endswith(".docx"):
+            try:
+                doc = Document(filename)
+                if len(doc.paragraphs) > 0:
+                    approaches.append("Approach 3: DOCX content extracted")
+            except Exception:
+                approaches.append("Approach 3: DOCX content extraction failed")
+
+        elif filename.endswith(".txt"):
+            try:
+                with open(filename, "r") as f:
+                    content = f.read(100)
+                    if content:
+                        approaches.append("Approach 3: TXT content validated")
+            except Exception:
+                approaches.append("Approach 3: TXT content validation failed")
+
+        #Select valid approaches
+        valid_approaches = [approach for approach in approaches if "matched" in approach or "extracted" in approach]
+        if len(valid_approaches) >= 2:
+            return file_type
+        else:
+            raise HTTPException(status_code=400, detail="File type mismatch, extraction failed.")
+
+    def extract_pdf_content(self, file_data):
+
+        try:
+            with pdfplumber.open(file_data) as pdf:
+                text = ""
+                for page in pdf.pages:
+                    text += page.extract_text()
+            return text
+        except Exception as e:
+            logging.error(f"Error extracting text from PDF: {e}")
+            raise HTTPException(status_code=400, detail="Failed to extract text from PDF")
+
+    def extract_docx_content(self, file_data):
+
+        try:
+            doc = Document(file_data)
+            text = ""
+            for para in doc.paragraphs:
+                text += para.text + "\n"
+            return text
+        except Exception as e:
+            logging.error(f"Error extracting text from DOCX: {e}")
+            raise HTTPException(status_code=400, detail="Failed to extract text from DOCX")
+
+    def extract_txt_content(self, file_data):
+
+        try:
+            text = file_data.decode('utf-8')
+            return text
+        except Exception as e:
+            logging.error(f"Error extracting text from TXT: {e}")
+            raise HTTPException(status_code=400, detail="Failed to extract text from TXT")
+
+    def extract_and_read_file(self, file_data, file_format):
+
+        if file_format == 'pdf':
+            return self.extract_pdf_content(file_data)
+        elif file_format == 'docx':
+            return self.extract_docx_content(file_data)
+        elif file_format == 'txt':
+            return self.extract_txt_content(file_data)
+        else:
+            raise HTTPException(status_code=400, detail="Unsupported file format")
+
+    def process_document(self):
```

Comment:
This function should take arguments customer_guid and filename
you should use the download function in minio_manager to download the file and then start processing the document
Once the file is parsed and the text document is retrieved, you should save it back in the same bucket with the filename with extension as filename.extension.rawcontent